core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -104,6 +108,52 @@ class HistoryServer(
 
   initialize()
 
+  private[spark] class HistoryServlet(val appId: String) extends HttpServlet {
```

Comment:
I think this servlet will go away when I merge this with the JSON end point change. I'm not sure whether I should add this to the JSON endpoint change or leave this servlet as is (since this is not really a JSON output, and from the current state it looks like the endpoints seem specific to returning JSON).